src/loaders/ncbi_downloader/gtdb.py

```diff
@@ -0,0 +1,311 @@
+    genome_ids = list()
+
+    taxonomy_urls = _form_gtdb_taxonomy_file_url(release_ver)
+
+    for taxonomy_url in taxonomy_urls:
+        # download GTDB taxonomy file to work_dir
+        url = urlparse(taxonomy_url)
+        taxonomy_file = os.path.join(work_dir, os.path.basename(url.path))
+        request.urlretrieve(taxonomy_url, taxonomy_file)
+
+        # parse genome id from first column of GTDB taxonomy tsv file
+        with open(taxonomy_file, "r") as f:
+            genome_ids.extend(
+                [parse_genome_id(line.strip().split("\t")[0]) for line in f]
+            )
+
+    return genome_ids, taxonomy_urls
+
+
+def _make_work_dir(root_dir: str, source_data_dir: str, source: str, env: str) -> str:
+    # make working directory for a specific collection under root directory
+    work_dir = os.path.join(root_dir, source_data_dir, source, env)
+    os.makedirs(work_dir, exist_ok=True)
+    return work_dir
+
+
+def _process_genome_ids(
+    work_dir: str,
+    genome_ids_unprocssed: list[str],
+    target_file_ext: list[str],
+    exclude_name_substring: list[str],
+    overwrite: bool = False,
+) -> list[str]:
+    """
+    Helper function that processes genome ids to avoid redownloading files.
+    """
+    if overwrite:
+        return genome_ids_unprocssed
+
+    genome_ids = list()
+    target_ext_count = len(target_file_ext)
+    for genome_id in genome_ids_unprocssed:
+        data_dir = os.path.join(work_dir, genome_id)
+        if not os.path.exists(data_dir):
+            genome_ids.append(genome_id)
+            continue
+
+        ext_count = 0
+        for file_name in os.listdir(data_dir):
+            if any([file_name.endswith(ext) for ext in target_file_ext]) and all(
+                [substring not in file_name for substring in exclude_name_substring]
+            ):
+                ext_count += 1
+
+        if ext_count != target_ext_count:
+            genome_ids.append(genome_id)
+
+    return genome_ids
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="PROTOTYPE - Download genome files from NCBI FTP server.",
+        formatter_class=loader_helper.ExplicitDefaultsHelpFormatter,
+    )
+
+    gtdb_release_vers = _parse_gtdb_release_vers()  # available GTDB release versions
+
+    required = parser.add_argument_group("required named arguments")
+    optional = parser.add_argument_group("optional arguments")
+
+    # Required flag argument
+    required.add_argument(
+        "--download_file_ext",
+        required=True,
+        type=str,
+        nargs="+",
+        help="Download only files that match given extensions",
+    )
+    required.add_argument(
+        "--release_ver",
+        required=True,
+        type=str,
+        choices=gtdb_release_vers,
+        help="GTDB release version",
+    )
+
+    # Optional argument
+    optional.add_argument(
+        f"--{loader_common_names.KBASE_COLLECTION_ARG_NAME}",
```

Comment:
Since we isolate gtdb downloading process, do we still need `kbase_collection` argument? Will we ever have a case where this won't be 'GTDB' if we call `gtdb.py`?

Reply:
I have a question around that here: https://kbase-jira.atlassian.net/browse/RE2022-209?focusedCommentId=91635

Namely, we're going to have 2 similar data sets - the data that was downloaded from NCBI using the GTDB IDs, which lives in `sourcedata/NCBI` and `collectionssource/<env>/<GTDB?>`, and the data that was loaded to the workspace, which lives in `sourcedata/WS/<env>` (I think) and `collectionssource/<env>/<GTDB?>`.

Those `<GTDB?>`s probably need to be different

Reply:
`collectionssource/<env>/GTDB_LOADED` makes sense to me.

Reply:
Actually, it's not a problem, since they'll have different <env> values

As such, to my mind we could hard code the GTDB collection name here